app/internal/notification.py

```diff
@@ -0,0 +1,126 @@
+from operator import attrgetter
+from typing import List, Union
+
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.orm import Session
+
+from app.database.models import (
+    Invitation,
+    Message,
+    InvitationStatusEnum,
+    MessageStatusEnum,
+)
+from app.internal.utils import create_model
+
+NOTIFICATION_TYPE = Union[Invitation, Message]
+
+UNREAD = [
+    InvitationStatusEnum.UNREAD,
+    MessageStatusEnum.UNREAD,
+]
+
+HISTORY = [
+    InvitationStatusEnum.DECLINED,
+    MessageStatusEnum.READ,
+]
+
+
+async def get_message_by_id(
+    message_id: int,
+    session: Session,
+) -> Union[Message, None]:
+    """Returns an invitation by an id.
+    if id does not exist, returns None."""
+    return session.query(Message).filter_by(id=message_id).first()
+
+
+def _is_unread(notification: NOTIFICATION_TYPE) -> bool:
+    """Returns True if notification is unread, False otherwise."""
+    return notification.status in UNREAD
+
+
+def _is_history(notification: NOTIFICATION_TYPE) -> bool:
+    """Returns True if notification should be
+    in history page, False otherwise.
+    """
+    return notification.status in HISTORY
+
+
+def get_unread_notifications(
+    session: Session,
+    user_id: int,
+) -> List[NOTIFICATION_TYPE]:
+    """Returns all unread notifications."""
+    return list(filter(_is_unread, get_all_notifications(session, user_id)))
```

Comment:
you can pass the filter method as param and then you need only you function (and not both get_unread_notifications and get_history_notifications)

Reply:
Thank's

Reply:
I tried to implement it, but i got confused.
can you explain again what you meant?

Reply:
You can pass the function as parameter to the function, something like:
```suggestion
    return list(filter(function_param, get_all_notifications(session, user_id)))
```

Then you don't need both get_unread_notifications and get_archived_notifications - you can keep one implementation.
Hope this helps